Simplenote/src/main/java/com/automattic/simplenote/PreferencesFragment.java

```diff
@@ -1,39 +1,49 @@
 package com.automattic.simplenote;
 
 
-import android.content.DialogInterface;
-import android.content.SharedPreferences;
-import android.os.AsyncTask;
-import android.support.v7.app.AlertDialog;
 import android.app.Fragment;
+import android.content.DialogInterface;
 import android.content.Intent;
+import android.content.SharedPreferences;
 import android.net.Uri;
+import android.os.AsyncTask;
 import android.os.Bundle;
+import android.support.v7.app.AlertDialog;
 import android.support.v7.preference.ListPreference;
 import android.support.v7.preference.Preference;
 import android.support.v7.preference.PreferenceManager;
 import android.widget.Toast;
 
 import com.automattic.simplenote.analytics.AnalyticsTracker;
 import com.automattic.simplenote.models.Note;
+import com.automattic.simplenote.models.Preferences;
+import com.automattic.simplenote.utils.HtmlCompat;
 import com.automattic.simplenote.utils.PrefUtils;
 import com.automattic.simplenote.utils.ThemeUtils;
 import com.simperium.Simperium;
 import com.simperium.android.LoginActivity;
 import com.simperium.client.Bucket;
+import com.simperium.client.BucketObjectMissingException;
+import com.simperium.client.BucketObjectNameInvalid;
 import com.simperium.client.User;
 import com.takisoft.fix.support.v7.preference.PreferenceFragmentCompat;
 import com.takisoft.fix.support.v7.preference.SwitchPreferenceCompat;
 
 import java.lang.ref.WeakReference;
 
+import static com.automattic.simplenote.models.Preferences.PREFERENCES_OBJECT_KEY;
+
 /**
  * A simple {@link Fragment} subclass.
  */
-public class PreferencesFragment extends PreferenceFragmentCompat implements User.StatusChangeListener, Simperium.OnUserCreatedListener {
+public class PreferencesFragment extends PreferenceFragmentCompat implements User.StatusChangeListener,
+        Simperium.OnUserCreatedListener {
 
     private static final String WEB_APP_URL = "https://app.simplenote.com";
 
+    private Bucket<Preferences> mPreferencesBucket;
+    private SwitchPreferenceCompat mAnalyticsSwitch;
```

Comment:
`SwitchPreferenceCompat` will go away with the update to Android 27. We can switch it to the Android support version.

See SwitchPreferenceCompat (2017-08-04) in [Changelog](https://github.com/Gericop/Android-Support-Preference-V7-Fix/blob/b628c9cecf175ea24789a9877613dff14bfb40cd/CHANGELOG.md)